great_expectations/datasource/fluent/interfaces.py

```diff
@@ -749,61 +765,92 @@ def __repr__(self) -> str:
         return self.data.__repr__()
 
 
-class Batch(FluentBaseModel):
+@public_api
+class Batch:
     """This represents a batch of data.
 
     This is usually not the data itself but a hook to the data on an external datastore such as
     a spark or a sql database. An exception exists for pandas or any in-memory datastore.
     """
 
-    datasource: Datasource
-    data_asset: DataAsset
-    batch_request: BatchRequest
-    data: BatchData
-    id: str = ""
+    # Immutable Attributes
+    _datasource: Datasource
+    _data_asset: DataAsset
+    _batch_request: BatchRequest
+    _data: BatchData
+    _id: str
```

Comment:
Do we need these class attrs if we aren't using Pydantic? I'd imagine just the constructor would be necessary